indico/modules/rb/controllers/user/reservations.py

```diff
@@ -457,9 +457,30 @@ def _checkParams(self):
     def _get_view(self, **kwargs):
         return RHRoomBookingNewBookingSimple._get_view(self, clone_booking=self._reservation, **kwargs)
 
+    def _update_datetime(self):
+        """Make necessary changes to reservation's start and end datetime.
+
+        Move the start date to the current day, adjust the end date
+        accordingly and if the start datetime is still in the past, change the
+        start and end time to the current (rounded up) time.
+
+        :return: A tuple with the new start_dt, end_dt
+        """
+        reservation_duration = self._reservation.end_dt - self._reservation.start_dt
+        date_delta = date.today() - self._reservation.start_dt.date()
+        start_dt = self._reservation.start_dt + date_delta
+        end_dt = start_dt + reservation_duration
+        if start_dt < datetime.now():
+            new_start_dt = round_up_to_minutes(datetime.now(), 15) + timedelta(minutes=15)
+            time_delta = new_start_dt - start_dt
+            start_dt = new_start_dt
+            end_dt = end_dt + time_delta
+        return start_dt, end_dt
```

Comment:
If you return a dict here you can simply use `changes.update(self._update_datetime())`